hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/DiscoveryJoiner.java

```diff
@@ -76,10 +78,29 @@ public DiscoveryJoiner(Node node, DiscoveryService discoveryService, boolean use
         for (DiscoveryNode discoveryNode : discoveredNodes) {
             Address discoveredAddress = usePublicAddress ? discoveryNode.getPublicAddress() : discoveryNode.getPrivateAddress();
             if (localAddress.equals(discoveredAddress)) {
+                if (!usePublicAddress && discoveryNode.getPublicAddress() != null) {
+                    // enrich member with client public address
+                    localMember.getAddressMap().put(EndpointQualifier.resolve(ProtocolType.CLIENT, "public"),
```

Comment:
As MC uses address map for advanced networking support and matching some metrics, this change may affect it.

@emre-aydin it may be a good idea to double check if this change doesn't break anything in MC. WDYT?

Reply:
Funny thing that you mention it right when I am troubleshooting a bug in this area and was just about to use `Member#getSocketAddress(EndpointQualifier.CLIENT)` for fixing it 😄  

Good idea to try changes in this branch with MC. Will add to my todo. Thanks for the ping.